roboteam_ai/src/control/ControlUtils.cpp

```diff
@@ -93,18 +105,11 @@ bool ControlUtils::hasClearVision(int fromID, int towardsID, roboteam_msgs::Worl
     return true;
 }
 
+
 /// Get the distance from PointToCheck towards a line, the line is not infinite.
 double ControlUtils::distanceToLineWithEnds(Vector2 pointToCheck, Vector2 lineStart, Vector2 lineEnd) {
-    Vector2 n = lineEnd - lineStart;
-    Vector2 pa = lineStart - pointToCheck;
-    Vector2 c = n*(n.dot(pa)/n.dot(n));
-    Vector2 d = pa - c;
-    Vector2 A = (pointToCheck - lineStart).project(lineStart, lineEnd);
-    Vector2 B = (pointToCheck - lineEnd).project(lineEnd, lineStart);
-    if ((A.length() + B.length()) > n.length()) {
-        return fmin(pa.length(), (lineEnd - pointToCheck).length());
-    }
-    else return d.length();
+    Vector2 projectionPoint = pointToCheck.project(lineStart,lineEnd);
+    return (projectionPoint - pointToCheck).length();
```

Comment:
@rolfvdhulst could you quickly look at this? I changed it so that it works for me, but I think you are actually using this code (and know what it does)?

Reply:
Well as is, this is wrong, as this will extend your line indefinitely  through the points lineStart and linEnd, and don't work with the line segment (so this is just distance to line). The previous version was also broken. There is a working version in the defensiveCoach branch.

I suggest you put a ToDo /warning here for now and that we merge it through another PR (I think @mrlukasbos has the same working one on another branch too?)